### DISCUSSION TITLE
branch filter pattern based on repository_owner

### DISCUSSION THREAD

**Discussion by sgn** [community/community#27012] [Actions]
<p>Hi GitHub Team,</p>
<p>I would like to trigger GitHub Actions for:</p>
<ul>
<li>only specific integration branches on main repository and maintainer’s repository, let’s say <strong>master</strong> , <strong>develop</strong> , and <strong>testing</strong>
</li>
<li>all branches except those whose name ends with <strong>-wip</strong> for all forks</li>
</ul>
<p>Is it possible? I’ve something similar to tried this:</p>
<pre><code>filter:
  true: [master, develop, testing]
  false: [' **', '!** -wip']

on:
  pull_request:
  push:
    branches: ${{ filter.${{ github.repository_owner == 'my-main' || github.repository_owner == 'maintainer-id' }} }}
    tags:
      - '**'
      - '! **wip**'
</code></pre>

**Comment by Yanjingzhu** [comment]
<p>Hi <a class="mention" href="/sgn">@sgn</a> , </p>
<p>The syntax you provided is not supported . And the idea of getting repository_owner is not working. </p>
<p>There is a GITHUB_REPOSITORY in workflow <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables" rel="nofollow noopener">default environment variables</a>, it always shows the main repository not the forked repo. </p>
<p>When a pull_request from a forked repo changed, there will be an event happened in the upstream repo (main repo) which could trigger a workflow in main repository.  </p>
<p>It is not that forks themselves can run arbitrary workflows,  it is that PRs raised from forks into the upstream will get run . </p>
<p>There is a previous ticket with similar question: <a href="https://github.community/t/why-cant-a-self-hosted-runner-verify-repo-owner-before-running/17007">https://github.community/t5/GitHub-Actions/Why-can-t-a-self-hosted-runner-verify-repo-owner-before-running/td-p/39261</a></p>

**Comment by sgn** [comment, marked as answer]
<p>Thanks,</p>
<p>We decided to have a workaround and check on</p>
<pre><code>jobs.&lt;id&gt;.if
</code></pre>
<p>instead.</p>
